spec/dummy/spec/helpers/react_on_rails_helper_spec.rb

```diff
@@ -10,24 +10,71 @@
     }
   end
 
-  describe "#sanitized_props_string(props)" do
-    let(:hash) do
-      {
-        hello: "world",
-        free: "of charge",
-        x: "</script><script>alert('foo')</script>"
-      }
-    end
+  let(:hash) do
+    {
+      hello: "world",
+      free: "of charge",
+      x: "</script><script>alert('foo')</script>"
+    }
+  end
 
-    let(:hash_sanitized) do
-      '{"hello":"world","free":"of charge","x":"\\u003c/script\\u003e\\u003cscrip'\
+  let(:hash_sanitized) do
+    '{"hello":"world","free":"of charge","x":"\\u003c/script\\u003e\\u003cscrip'\
       "t\\u003ealert('foo')\\u003c/script\\u003e\"}"
+  end
+
+  let(:hash_unsanitized) do
+    "{\"hello\":\"world\",\"free\":\"of charge\",\"x\":\"</script><script>alert('foo')</script>\"}"
+  end
+
+  describe "#json_safe_and_pretty(hash_or_string)" do
+    let(:special_characters_hash) { { tested_characters: '<>&\u2028\u2029' } }
+
+    let(:special_characters_sanitized_hash) do
+      %({"tested_characters":"\\u003c\\u003e\\u0026\\\\u2028\\\\u2029"})
+    end
+
+    it "should raise an error if not hash nor string passed" do
+      expect { helper.json_safe_and_pretty(false) }.to raise_error
     end
 
-    let(:hash_unsanitized) do
-      "{\"hello\":\"world\",\"free\":\"of charge\",\"x\":\"</script><script>alert('foo')</script>\"}"
+    it "converts a hash to escaped JSON" do
+      escaped_json = helper.json_safe_and_pretty(hash)
+      expect(escaped_json).to eq(hash_sanitized)
     end
 
+    it "converts a string to escaped JSON" do
+      escaped_json = helper.json_safe_and_pretty(hash_unsanitized)
+      expect(escaped_json).to eq(hash_sanitized)
+    end
+
+    shared_examples :escaped_json do
+      it "returns a well-formatted json with escaped characters" do
+        expect(helper.json_safe_and_pretty(special_characters_hash))
+          .to eq(special_characters_sanitized_hash)
+      end
+    end
+
+    context 'with Rails 3' do
+      before { allow(Rails).to receive(:version).and_return(3) }
```

Comment:
I'm not really sure that mocking the version is the better way to do. 

Could it be a good idea to configure Travis to lauch the tests with differents Rails version? I'm just raising the question, I know this imply to extend the tests duration and to change the Travis configuration.